apps/shipments/views/shipment_overview.py

```diff
@@ -16,49 +16,85 @@
 
 import logging
 
-from django.conf import settings
+from django.contrib.gis.geos import Polygon
+from django.db.models import Max
+from django_filters.rest_framework import DjangoFilterBackend
 from influxdb_metrics.loader import log_metric
-from rest_framework import permissions, status, renderers
-
+from rest_framework import permissions, status
+from rest_framework import filters
+from rest_framework.generics import ListAPIView
 from rest_framework.response import Response
-from rest_framework.views import APIView
-from shipchain_common.exceptions import RPCError
-from shipchain_common.pagination import CustomResponsePagination
-
 
 from apps.permissions import get_owner_id
-
-from ..iot_client import DeviceAWSIoTClient
-from ..serializers import DevicesQueryParamsSerializer
+from ..serializers import DevicesQueryParamsSerializer, ShipmentLocationSerializer
+from ..models import Shipment, TrackingData
+from ..filters import ShipmentFilter, SHIPMENT_SEARCH_FIELDS
 
 LOG = logging.getLogger('transmission')
 
 
-class ListDevicesStatus(APIView):
-    http_method_names = ['get', ]
+class ShipmentOverviewListView(ListAPIView):
+
     permission_classes = (permissions.IsAuthenticated, )
-    pagination_class = CustomResponsePagination
-    renderer_classes = (renderers.JSONRenderer,)
+
+    serializer_class = ShipmentLocationSerializer
+
+    filter_backends = (filters.SearchFilter, filters.OrderingFilter, DjangoFilterBackend,)
+
+    filterset_class = ShipmentFilter
+
+    ordering_fields = ('created_at', )
+
+    search_fields = SHIPMENT_SEARCH_FIELDS
+
+    queryset = TrackingData.objects.filter(shipment__device_id__isnull=False)
+    shipment_queryset = Shipment.objects.filter(device_id__isnull=False)
+
+    def filter_tracking_data_queryset(self, owner_id, queryset, query_params):
+
+        # Shipment owner filter
+        shipment_queryset = self.shipment_queryset.filter(owner_id=owner_id)
+
+        # Shipment fields filter
+        shipment_queryset = self.filter_queryset(shipment_queryset)
+
+        latest_tracking = shipment_queryset.annotate(
+            latest_tracking_created=Max('trackingdata__created_at')
+        ).values_list('latest_tracking_created')
+
+        tracking_data_queryset = queryset.filter(shipment__in=shipment_queryset,
+                                                 created_at__in=latest_tracking)
+
+        bbox_tracking_data_queryset = self.bbox_filter(tracking_data_queryset, query_params.get('in_bbox'))
+
+        return bbox_tracking_data_queryset
+
+    @staticmethod
+    def bbox_filter(queryset, bbox_param):
```

Comment:
Consider using the existing bbox filter in the DRF-gis package: https://github.com/openwisp/django-rest-framework-gis#inbboxfilter

Reply:
This library is going to be very useful, Thanks! :+1: